### DISCUSSION TITLE
Use other Databases

### DISCUSSION THREAD

**Discussion by ebaer-payix** [hagopj13/node-express-boilerplate#82] [Ideas]
Would it be relatively easy to replace out Mongoose/MongoDB with another ORM/Database ?
ie: PostgreSQL, or MariaDB

I know you have done a lot of work here, and I would love to leverage it, but I need to use a different database.

Thanks

**Comment by hagopj13** [comment]
Hi @ebaer-payix, you can of course replace the DB and handle the code changes at the right places. You will have to do most of the changes at the `model` and `services` levels. Should be a straightforward process as the code is quite modular.

If you're looking for a good ORM, I suggest you look at https://typeorm.io/

**Comment by ebaer-payix** [comment]
Hey Hagop,

Thanks for responding.

I actually went through the pain of converting it to MySQL and Prisma.
It is not 100% converted in the branch I saved before I starting building my application, but it is pretty close.

If you are interested, I can provide you with a copy of that branch, that you, or someone else could complete in a way that you can publish it again.

BTW, I tried to keep the logic in place to support Mongoose or MySQL, along with adding Basic Authentication or Token Authentication to the mix, which is controllable via the .env file.

Any interest ?

Thanks

ewb

________________________________

**Comment by fernandocanizo** [comment]
I think they should be separate repositories. In my opinion, the necessary work to produce an installer that cleans unrelated stuff according to which database you'll want to use doesn't worth the time. Yesterday night I started a repo based on Hagop work to support Postgresql instead of mongo. Currently is all broken, cause I started by adapting some things to my liking and still haven't touched the database, but I expect to have it working soon. You can check it at: [neps](https://github.com/fernandocanizo/node-express-postgres-scaffold). As you can see I dropped the installer right away. In my opinion, being this a starting point for developers and not for clueless end-users, I think is enough with a set of steps in the README instead of maintaining a script to save three commands on the CLI.

**Comment by ebaer-payix** [comment]
Fernando,

Thanks for the follow up.
My offer still stands.
I you would like a copy of the repo/branch with the MySQL / Prisma changes, I will provide it to you .
If you think it needs to be in a separate repo, I think that is fine.
If you want to setup that repo, and give me access, perhaps later I/we can work through any bugs we find ?

Just let me know...

Thanks

ewb

________________________________

**Reply by fernandocanizo** [reply]
Sure, I need to start messing with the database change before, cause I've been toying so far. But maybe all SQL DBs could be put into the same repo. I'll be in touch.

**Comment by trasherdk** [comment]
I was looking at that https://typeorm.io/ thing that actually have support for MongoDB.
Maybe that is the route to follow, adding support for alternative databases.

I, myself, think MongoDB is only good for example Hello World or To-Do apps.
